test/common/config/grpc_mux_impl_test.cc

```diff
@@ -336,6 +337,70 @@ TEST_F(GrpcMuxImplTest, TooManyRequests) {
                           onReceiveMessage(1));
 }
 
+//  Verifies that a messsage with no resources is accepted.
+TEST_F(GrpcMuxImplTest, UnwatchedTypeAcceptsEmptyResources) {
+  EXPECT_CALL(*async_client_, start(_, _)).WillOnce(Return(&async_stream_));
+
+  const std::string& type_url = Config::TypeUrl::get().ClusterLoadAssignment;
+
+  grpc_mux_->start();
+  {
+    // subscribe and unsubscribe to simulate a cluster added and removed
+    expectSendMessage(type_url, {"y"}, "");
+    auto temp_sub = grpc_mux_->subscribe(type_url, {"y"}, callbacks_);
+    expectSendMessage(type_url, {}, "");
+  }
+
+  // simulate the server sending empty CLA message to notify envoy that the CLA was removed.
+  std::unique_ptr<envoy::api::v2::DiscoveryResponse> response(
+      new envoy::api::v2::DiscoveryResponse());
+  response->set_nonce("bar");
+  response->set_version_info("1");
+  response->set_type_url(type_url);
+
+  // This contains zero resources. No discovery request should be sent.
+  grpc_mux_->onReceiveMessage(std::move(response));
+
+  // when we add the new subscription version should be 1 and nonce should be bar
+  expectSendMessage(type_url, {"x"}, "1", "bar");
+
+  // simulate a new cluster x is added. add CLA subscription for it.
+  auto sub = grpc_mux_->subscribe(type_url, {"x"}, callbacks_);
+  expectSendMessage(type_url, {}, "1", "bar");
+}
+
+//  Verifies that a messsage with some resources is rejected when there are no watches.
+TEST_F(GrpcMuxImplTest, UnwatchedTypeRejectsResources) {
+  EXPECT_CALL(*async_client_, start(_, _)).WillOnce(Return(&async_stream_));
+
+  const std::string& type_url = Config::TypeUrl::get().ClusterLoadAssignment;
+
+  grpc_mux_->start();
+  {
```

Comment:
Nit: Don't think you need to use an explicit scope here, since we're not binding anything.